packages/frontend/editor-ui/src/features/dataStore/composables/useDataStorePagination.ts

```diff
@@ -0,0 +1,49 @@
+import { ref } from 'vue';
+
+export type UseDataStorePaginationOptions = {
+	initialPage?: number;
+	initialPageSize?: number;
+	pageSizeOptions?: number[];
+	onChange?: (page: number, pageSize: number) => Promise<void> | void;
+};
+
+export const useDataStorePagination = (options: UseDataStorePaginationOptions = {}) => {
+	const currentPage = ref<number>(options.initialPage ?? 1);
+	const pageSize = ref<number>(options.initialPageSize ?? 20);
+	const totalItems = ref<number>(0);
+	const pageSizeOptions = options.pageSizeOptions ?? [10, 20, 50];
+
+	const setTotalItems = (count: number) => {
+		totalItems.value = count;
+	};
+
+	const setCurrentPage = async (page: number) => {
```

Comment:
Maybe pushing this too far :D but, since we have max page size 50 on back-end, maybe we can create a type that limits the range for this number (something like [this](https://stackoverflow.com/questions/39494689/is-it-possible-to-restrict-number-to-a-certain-range/70307091#70307091))?

Reply:
Ah sorry, this doesn't make sense but should be nice not to allow numbers `< 1`.